var/spack/repos/builtin/packages/variorum/package.py

```diff
@@ -29,33 +33,42 @@ class Variorum(CMakePackage):
     # Variants #
     ############
     variant("shared", default=True, description="Build Variorum as shared lib")
-    variant("docs", default=False, description="Build Variorum's documentation")
     variant(
         "build_type",
         default="Release",
         description="CMake build type",
-        values=("Debug", "Release"),
+        values=("Debug", "Release", "RelWithDebInfo"),
     )
+    #intel gpu not supported in spack
+    #variant("intel_gpu", default=False, description="Build for Intel GPU architecture")
+
+    for value in CudaPackage.cuda_arch_values:
+        if value != "70":
+            conflicts(f"cuda_arch={value}")
+
+    for value in ROCmPackage.amdgpu_targets:
+        if value != "gfx906" and value != "gfx906:xnack-":
+            conflicts(f"amdgpu_target={value}")
 
```

Comment:
```suggestion
    supported_amd_targets = ["zen2"]
    supported_arm_targets = ["neoverse_n1"]
    supported_ibm_targets = ["power9le"]
    supported_intel_targets = [
        "sandybridge",
        "ivybridge",
        "haswell",
        "broadwell",
        "skylake",
        "cascadelake",
        "icelake",
    ]
    for arch in archspec.cpu.TARGETS:
        if arch not in supported_amd_targets + supported_arm_targets + supported_ibm_targets + supported_intel_targets:
            conflicts(f"target={target}")
```

Then below in the `cmake_args` method you can just check the vendor.

If the non-supported targets are a valid build target when building for GPU, you can switch the conflicts statement above to `conficts(f"target={target}", when="~cuda~rocm")`